client/verta/verta/_internal_utils/_utils.py

```diff
@@ -259,19 +260,55 @@ def make_request(method, url, conn, **kwargs):
     with requests.Session() as s:
         s.mount(url, HTTPAdapter(max_retries=conn.retry))
         try:
-            response = s.request(method, url, **kwargs)
+            request = requests.Request(method, url, **kwargs).prepare()
+            response = s.send(request, allow_redirects=False)
+
+            # store initial response to add back into final history
+            initial_response = response
+
+            # manually inspect redirects to stop on 302s
+            responses = itertools.chain([initial_response], s.resolve_redirects(initial_response, request))
```

Comment:
I don't understand what this is doing.

Reply:
`s.resolve_redirects(initial_response, request)` will yield the sequence of redirects that occur _after_ the initial response for the `for` loop below.
But I need to check the status code of that initial response too.

So I'm using `itertools.chain()` to yield the initial response for the first iteration of the loop, then proceed to iterate through `s.resolve_redirects()`.

Reply:
Got it